### PR TITLE
"Error writing cache count for transactions - Cache Disabled" when /api/node/status is called - Closes #2482

### LINKED ISSUE #2482: "Error writing cache count for transactions - Cache Disabled" when /api/node/status is called
### Expected behavior
No errors in the log when /api/node/status is called

### Actual behavior
When I call the /api/node/status endpoint, I get the following error in the log:
`Error writing cache count for transactions - Cache Disabled`

### Steps to reproduce
monitor the log with `tail -f logs/lisk.log`
then call
`curl -s localhost:7000/api/node/status`

### Which version(s) does this affect? (Environment, OS, etc...)
1.1.0.rc0 on testnet

### PR DESCRIPTION
**Pull request by pablitovicente**
### What was the problem?

- loader.js was enabling/disabling cache regardless of the cache being enabled/disabled

### How did I fix it?

- By adding extra checks in loader.js

### How to test it?

- Sync a node until loader tries to enable the cache and the cache config is off (or redis is turned off)
- `GET api/node/status` and you shouldn't see the error message in the console attempting to write/read from the cache

### Review checklist

* The PR resolves #2482
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: framework/src/modules/chain/submodules/loader.js [hunk 1]
```diff
@@ -867,7 +867,7 @@ __private.loadBlocksFromNetwork = function(cb) {
  */
 __private.sync = function(cb) {
 	library.logger.info('Starting sync');
-	if (components.cache.cacheReady) {
+	if (components.cache.options.enabled && components.cache.cacheReady) {
```
Comment (lsilvs) [inline framework/src/modules/chain/submodules/loader.js]: I would remove the second condition as it isn't necessary and creates confusion

```suggestion
	if (components.cache.options.enabled) {
```

File: framework/src/modules/chain/submodules/loader.js [hunk 2]
```diff
@@ -915,7 +915,7 @@ __private.sync = function(cb) {
 			__private.blocksToSync = 0;
 
 			library.logger.info('Finished sync');
-			if (!components.cache.cacheReady) {
+			if (components.cache.options.enabled && !components.cache.cacheReady) {
```
Comment (lsilvs) [inline framework/src/modules/chain/submodules/loader.js]: Same as above

```suggestion
			if (components.cache.options.enabled) {
```

File: framework/src/modules/chain/submodules/loader.js (hunk 1 above)
Comment (pablitovicente) [reply framework/src/modules/chain/submodules/loader.js]: Cache can be enabled and not ready (for example redis was not up at boot or went down)

File: framework/src/modules/chain/submodules/loader.js (hunk 2 above)
Comment (pablitovicente) [reply framework/src/modules/chain/submodules/loader.js]: Cache can be enabled and not ready (for example redis was not up at boot or went down)

**Comment by ManuGowda** [comment]
@pablitovicente do we have tests covered for this scenario?

**Comment by pablitovicente** [comment]
> @pablitovicente do we have tests covered for this scenario?

I think this scenario only happens after a full sync so I don't think it's practical to test from within SDK

**Pull request merged**